### DISCUSSION TITLE
How to add onDrop capabilities to the whole window

### DISCUSSION THREAD

**Discussion by axillv** [iPlug2/iPlug2#1122] [Q&A] [answered]
As the title suggests, I tried creating an IControl inherited class, that would handle onDrop. However, it blocks everything else for the controls below. In the same spirit, placing it at the bottom layer, it isn't accessible.

Is there a way to allow mouse event "passthroughs" or something similar? How would you go about adding onDrop capabilities to the whole plugin window?

Thanks!

**Comment by olilarkin** [comment]
I think we'll have to think about how we can provide a global drag and drop callback that doesn't necessarily link to a control.

FYI: here are some WIP improvements to DND: https://github.com/iPlug2/iPlug2/pull/1123/

**Comment by axillv** [comment]
I see, so as is right now, it's impossible to do that right? And if I understand correct, #1064 should provide that functionality.

**Comment by olilarkin** [comment, marked as answer]
I’m not sure it’s impossible right now and currently that pr doesn’t
address it. It might give you some ideas though how you can hack it for
your purposes. If you think what you do is good enough feel free to make a
PR

**Comment by axillv** [comment]
I would be interested in giving it a try, however I only just started using the framework? Is there anywhere I could start trying to figure this out? Perhaps a new IControl subclass with mouse event passthrough or something like that? I'd love to try!

**Comment by olilarkin** [comment]
I made an issue: https://github.com/iPlug2/iPlug2/issues/1126

I would probably take the approach that is used with the "key handler func" for receiving global key presses

https://github.com/search?q=repo%3AiPlug2%2FiPlug2%20keyHandlerFunc&type=code
